### ISSUE TITLE
Make callout invisible

### ISSUE THREAD

**Issue by fritz-hh** [asciidoctor/asciidoctor#2961]
Hi,

I faced the following issue with callouts and could not find out a nice was to solve it. I would be grateful for an advice.

I include the same source file twice in a page.
At 1 location, I would like to show the callouts. At the other location, the callout should not be displayed.
Is there a way to tell asciidoctor not to show callouts defined in a source block?

Thank you :-)

**Comment by elextr** [comment]
If you can substitute attributes in the block that is repeated you can do it like this:

```

:one: <1>
:two: <2>

[subs=attributes]
----
blah blah {one}
blah blah {two}
----

<1> blah
<2> blah


:one:
:two:

[subs=attributes]
----
blah blah {one}
blah blah {two}
----
```

**Comment by mojavelinux** [comment]
Another way would be to just let the callouts be included in both cases and then hide them using CSS. Add a role to the listing block like `hide-callouts` and then add a CSS rule to add `display: none` to the callout element in that case.

This is an interesting problem, though. We ran into the same thing in the Asciidoctor User Manual. Perhaps we could consider supporting an option to drop the callouts. It's not the same as removing the callouts substitution because you actually want to find them, but then get rid of them.

**Comment by mojavelinux** [comment]
What we need to determine is how to express the behavior you want. This is always the hardest part with every change / improvement to AsciiDoc.

**Comment by mojavelinux** [comment]
> don't you get warnings if callouts don't have list items referring to them?

Yes, though this case would be the other way around. There would be callouts with no list.

There's no doubt a judicious combination of subs and attributes could be used to remove the callouts, but I think we can shoot for something more elegant. So let's focus on that path.

**Comment by mojavelinux** [comment]
Having said that, CSS is probably the best short-term workaround.

**Comment by elextr** [comment]
> Having said that, CSS is probably the best short-term workaround.

Yes, so long as the user is aware they get the warnings until the proper solution is in place.

> So let's focus on that path.

Agree.

An option like `remove_callouts` seems sensible and obvious.  The frontend still parses the callouts if subs=callouts is set, but the backend now removes the callouts from the block and obviously does not complain about missing list items.

**Comment by elextr** [comment]
> The problem comes when you have a callout list but no callout numbers.

Ok, I read the warning backwards.  Getting a warning from callouts without list items can be a future enhancement.

**Comment by mojavelinux** [comment]
I also now understand why you were getting warnings in your example. It has to do with how Asciidoctor validates callouts. It does a pre-scan for callouts during parsing. Since the callouts are hidden by the attribute reference, the scanner doesn't find them and thus mistaking assumes they are missing. Just FYI.

**Comment by fritz-hh** [comment]
Thank you @mojavelinux and @elextr for your proposal and workarounds.

Having a `remove_callouts` option would be great !
I the mean time, I will try to see solve it using CSS (I also get warning when using substitutions)

**Comment by wimdeblauwe** [comment]
Is there any workaround if you are doing PDF export? I also would like to use the same source file for inclusion which has callouts, but I want in 1 case to have the callouts removed (and there would be nothing under the source block that references the callouts as well in that case).

**Comment by mojavelinux** [comment]
The only workaround currently is to either include a different snippet (which may not be possible) or to use a custom include processor that removes the callout before delegating to the built-in include processor. If you need help with that, please ask in the chat at https://chat.asciidoctor.org.
